### PR TITLE
L2-283: Proposal Details Page - My votes block

### PR DESCRIPTION
**Pull request by lmuntanerDfinity**
# Motivation

* Add user votes in proposal details

# Changes

* VotesCard reads neurons store and filters the ones that voted.
* VotesCard renders data about each neuron and vote.

# Tests

* Update VotesCard test.

### CODE REVIEW

**Comment by lmuntanerDfinity** [comment]
Screenshot

Screenshot 2022-03-03 at 16 13 15

File: frontend/svelte/env.config.mjs [hunk 1]
```diff
@@ -38,7 +38,7 @@ const LEDGER_CANISTER_ID = "ryjl3-tyaaa-aaaaa-aaaba-cai";
 // TODO: Move to index.html L2-265
 const OWN_CANISTER_ID =
   process.env.OWN_CANISTER_ID ||
-  (development ? "qhbym-qaaaa-aaaaa-aaafq-cai" : "qoctq-giaaa-aaaaa-aaaea-cai");
```
Comment (lmuntanerDfinity) [inline frontend/svelte/env.config.mjs:41]: Change because the canister id changed with a new testnet deployment.

We need to do https://dfinity.atlassian.net/browse/L2-265 to avoid this in the future.

**Comment by lmuntanerDfinity** [comment]
@peterpeterparker @mstrasinskis please review

File: frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte [hunk 2]
```diff
@@ -13,6 +18,29 @@
...
+    id: NeuronId;
+    votingPower: number;
+    vote: Vote;
+  };
+  const voteIconMapper = {
+    [Vote.NO]: IconThumbDown,
+    [Vote.YES]: IconThumbUp,
+  };
+  let neuronsVotedForProposal: CompactNeuronInfo[];
+  neuronsStore.subscribe((neurons) => {
```
Comment (mstrasinskis) [inline frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte:32]: `unsubscribe` or `$: ...`?

File: frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte [hunk 3]
```diff
@@ -13,6 +18,29 @@
...
+    neuronsVotedForProposal = votedNeurons({
+      neurons: neurons,
+      proposal: proposalInfo,
+    }).map(({ neuronId, recentBallots, votingPower }) => ({
+      id: neuronId,
+      votingPower: Number(votingPower) / E8S_PER_ICP,
```
Comment (mstrasinskis) [inline frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte:38]: Could you plz insert a comment before `// TODO: replace w/ formatVotingPower()`?

File: frontend/svelte/src/lib/stores/neurons.store.ts [hunk 4]
```diff
@@ -1,14 +1,16 @@
 import type { NeuronInfo } from "@dfinity/nns";
 import { writable } from "svelte/store";
 
+export type NeuronsStore = NeuronInfo[];
```
Comment (mstrasinskis) [inline frontend/svelte/src/lib/stores/neurons.store.ts:4]: 👍

**Review by mstrasinskis — CHANGES_REQUESTED** [review]
Looks good. Just a couple of comments.

File: frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte (hunk 2 above)
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte:32]: let me try again, it wasn't working, but maybe I wasn't using it properly.

File: frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte (hunk 3 above)
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte:38]: 👍

File: frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte (hunk 2 above)
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte]: I don't know what I was doing, but now it worked.

**Comment by lmuntanerDfinity** [comment]
@mstrasinskis ready for another review

**Review by mstrasinskis — APPROVED** [review]
LGTM

File: frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte [hunk 5]
```diff
@@ -13,6 +17,31 @@
   const yesValue = Number(yes) / E8S_PER_ICP;
   const noValue = Number(no) / E8S_PER_ICP;
   const summ = yesValue + noValue;
+
+  type CompactNeuronInfo = {
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte:21]: shouldn't this be an `interface` not a `type` @lmuntanerDfinity ?

Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte:21]: Do we have a preference? I felt I saw `type` being used for this. I don't mind.

Comment (peterpeterparker) [reply frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte:21]: I always use `interface` for object. Open to discussion. Can we have an eslint rule?

Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/proposal-detail/VotesCard.svelte:21]: Yes, we could.
